yarn-project/aztec-rpc/src/kernel_prover/kernel_prover.ts

```diff
@@ -95,15 +98,57 @@ export class KernelProver {
           `Too many items in the call stack. Maximum amount is ${PRIVATE_CALL_STACK_LENGTH}. Got ${privateCallStackPreimages.length}.`,
         );
       }
+      // Pad with with empty items to reach max/const length expected by circuit.
       privateCallStackPreimages.push(
         ...Array(PRIVATE_CALL_STACK_LENGTH - privateCallStackPreimages.length)
           .fill(0)
           .map(() => PrivateCallStackItem.empty()),
       );
 
-      const privateCallData = await this.createPrivateCallData(currentExecution, privateCallStackPreimages);
+      // TODO(dbanks12): https://github.com/AztecProtocol/aztec-packages/issues/779
+      // What if app circuit outputs different #read-requests vs what the RPC client
+      // gets from the simulator?
+      const readRequestMembershipWitnesses = [];
+      for (let rr = 0; rr < currentExecution.readRequestCommitmentIndices.length; rr++) {
+        if (currentExecution.callStackItem.publicInputs.readRequests[rr] == Fr.zero()) {
+          // TODO(dbanks12): is this needed?
+          // if rr is 0 somehow, that means it we have reached the last
+          // rr for this private call / kernel iteration
+          break;
+        }
+        //const leafIndex = currentExecution.readRequestCommitmentIndices[rr];
+        let leafIndex = currentExecution.readRequestCommitmentIndices[rr];
+        // WARNING/DANGER/FIXME
+        // TODO(dbanks12): the simulator oracle retrieves a leaf index from the DB
+        // that does not match its location in the private data tree.
+        // Debugging the cross chain messaging test led to the discovery
+        // that when leafIndex 5 is communicated from the simulator,
+        // the corresponding leaf is 0, but the commitment described by
+        // its read request lives at leaf index 17.
+        // https://github.com/AztecProtocol/aztec-packages/issues/788
+        leafIndex = leafIndex == 5 ? 17 : leafIndex;
```

Comment:
Scary hack for now.... If someone can help me debug this asap, great. This is a bug uncovered thanks to read requests.

Reply:
Resolved by Maddia's PR into this one